src/StepParityCost.cpp

```diff
@@ -0,0 +1,848 @@
+#include "global.h"
+#include "StepParityCost.h"
+#include "NoteData.h"
+#include "TechCounts.h"
+#include "GameState.h"
+	
+using namespace StepParity;
+
+
+template <typename T>
+bool isEmpty(const std::vector<T> & vec, int columnCount) {
+	for (int i = 0; i < columnCount; i++)
+	{
+		if(static_cast<int>(vec[i]) != 0)
+		{
+			return false;
+		}
+	}
+	return true;
+}
+
+
+float* StepParityCost::getActionCost(State * initialState, State * resultState, std::vector<Row>& rows, int rowIndex)
```

Comment:
Where is the allocated `costs` array that this function returns being deleted? I'd expect to see a `delete[]` call in `StepParityNode`'s destructor as it looks like `StepParityNode` is responsible for managing this memory.

Reply:
Yeah, it's in `StepParityNode`'s destructor: https://github.com/mjvotaw/itgmania/blob/5d5a72f7a21fe8cd14a3f215f539fd31886830ad/src/StepParityDatastructs.h#L291

But we don't actually need to hold onto all of the cost data, that was mainly just for debugging purposes. I'm planning on switching that to just store the total cost value.

Reply:
Oh my bad. I just realized my local copy is on what looks to be an old branch where the destructor wasn't added yet. I was on `step-parity-poc`.